store/models/job_spec_test.go

```diff
@@ -114,21 +145,39 @@ func TestTaskSpec_UnmarshalJSON(t *testing.T) {
 		json          string
 		output        string
 	}{
-		{"noop", "noop", 0, `{"type":"noOp"}`, `{"type":"noop","confirmations":0}`},
+		{"noop", "noop", 0,
+			`{"type":"noOp"}`,
+			`{"type":"noop","confirmations":0}`,
+		},
 		{
-			"httpget",
-			"httpget",
-			0,
+			"httpget", "httpget", 0,
 			`{"type":"httpget","url":"http://www.no.com"}`,
 			`{"type":"httpget","url":"http://www.no.com","confirmations":0}`,
 		},
-		{
-			"with confirmations",
-			"noop",
-			10,
+		{"with confirmations", "noop", 10,
 			`{"type":"noop","confirmations":10}`,
 			`{"type":"noop","confirmations":10}`,
 		},
+		{"with variations in key name casing for 'type'", "noop", 10,
+			`{"TYPE":"noop","confirmations":10}`,
+			`{"type":"noop","confirmations":10}`,
+		},
+		{"with variations in key name casing for 'confirmations'", "noop", 10,
+			`{"type":"noop","CONFIRMATIONS":10}`,
+			`{"type":"noop","confirmations":10}`,
+		},
+		{"with variations in key name casing for other keys", "noop", 10,
+			`{"type":"noop","CONFIRMATIONS":10,"foo":"bar","Foo":"baz","FOO":3}`,
+			`{"type":"noop","confirmations":10,"foo":"bar","Foo":"baz","FOO":3}`,
```

Comment:
If we were to use this output, then pass this to rust's serde json parser, it would pick `"foo":"bar"` whereas the go code would pick `"FOO":3` right?

Reply:
I think so.

As I understand it, you are right when we have a `foo` field:

```
type FooAdapter struct {
    Foo int `json:"foo"`
}
```

A concrete example for the multiplier adapter:

```
{
  "type":"multiply",
  "confirmations":10,
  "times":1,
  "TIMES":9
}
```

Rust has 1 and Golang has 9.

To summarize, we still preserve all the Foos, but the subsequent interpretation of it varies based on the run time deserialization logic of the adapter. This doesn't matter for external adapters, but since rust deserialization != golang deserialization, this will be an issue.

Let me know if I'm off. Just wanted to add clarity to @j16r's question. After seeing this, I'm beginning to form the opinion that we shouldn't allow duplicate case insensitive keys. Just error out from the get go.